code/go/0chain.net/smartcontract/dbs/event/process.go

```diff
@@ -698,6 +705,16 @@ func (edb *EventDb) addStat(event Event) (err error) {
 		return edb.addOrUpdateChallengePools(*cps)
 	case TagCollectProviderReward:
 		return edb.collectRewards(event.Index)
+	case TagMinerHealthCheck:
+		return edb.ProviderHealthCheck(spenum.Miner, event.Index, event.Data)
```

Comment:
Please check how the other events with merge are processed, such as `TagUpdateBlobberStat`, basically the `event.Data` will become a type of slice with all merged data from events.